compiler/utils.go

```diff
@@ -319,9 +328,30 @@ func isVarOrConst(o types.Object) bool {
 }
 
 func isPkgLevel(o types.Object) bool {
-	return o.Parent() != nil && o.Parent().Parent() == types.Universe
+	// Note: named types are always assigned a variable at package level to be
+	// initialized with the rest of the package types, even the types declared
+	// in a statement inside a function.
```

Comment:
Is it worth explaining why?

Reply:
It is, but alas I don't know the answer :)